packages/@coorpacademy-components/src/template/skill-detail/all-courses.js

```diff
@@ -0,0 +1,181 @@
+import React, {useState, useCallback, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {get, getOr, sortBy} from 'lodash/fp';
+import Provider from '../../atom/provider';
+import Select, {SelectOptionPropTypes} from '../../atom/select';
+import ButtonLink from '../../atom/button-link';
+import SearchForm from '../../molecule/search-form';
+import CardsGrid from '../../organism/cards-grid';
+import searchValueIncluded from '../../util/search-value-included';
+import InputSwitch from '../../atom/input-switch';
+import style from './all-courses.css';
+
+const FilterButton = (props, context) => {
+  const {selected, filter, onClick} = props;
+  const {skin} = context;
+  const primarySkinColor = get('common.primary', skin);
+
+  const buttonProps = {
+    customStyle: {
+      backgroundColor: selected ? primarySkinColor : '#E1E1E3',
+      color: selected ? '#FFFFFF' : '#515161',
+      transition: 'background-color 0.15s ease-in-out, color 0.15s ease-in-out',
+      width: 'fit-content'
+    },
+    label: filter,
+    onClick,
+    'data-name': 'filter-type-course-button'
+  };
+
+  return <ButtonLink {...buttonProps} />;
+};
+
+FilterButton.contextTypes = {
+  skin: Provider.childContextTypes.skin,
+  translate: Provider.childContextTypes.translate
+};
+
+FilterButton.propTypes = {
+  selected: PropTypes.bool,
+  filter: PropTypes.string,
+  onClick: PropTypes.func
+};
+
+const AllCourses = (props, context) => {
+  const {skillIncludedCourses, totalCourses, filters, sorting} = props;
+  const {options, onChange} = filters;
+  const {list, loading} = skillIncludedCourses;
+  const {translate} = context;
+  const [showCompleted, setShowCompleted] = useState(true);
+  const [searchValue, setSearchValue] = useState('');
+  const [searchResults, setSearchResults] = useState(
+    sortBy(course => -getOr(0, ['progress'], course), list)
```

Comment:
indeed you can't just `-progress` (this will just ignore entry without progress)

(note you could have path `progress` (no need for single element array))